substrate/frame/sudo/src/extension.rs

```diff
@@ -110,6 +111,7 @@ where
 		Ok((
 			ValidTransaction {
 				priority: info.total_weight().ref_time() as TransactionPriority,
+				provides: vec![(who, call).encode()],
```

Comment:
dq: is call guaranteed to be unique here?
Otherwise we won't be able to send numerous sudo calls. I can imagine someone calling same sudo operation few times (e.g. transfer funds).

Reply:
AFAIU it is not, but for chains that also have the `CheckNonce` extension there is another unique tag provided. A chain with only the `CheckOnlySudoAccount` extension but without `CheckNonce` will have to work around this uniqueness guarantee (e.g., batch transactions).

Reply:
> dq: is call guaranteed to be unique here?
> Otherwise we won't be able to send numerous sudo calls

Fair enough. Does it help, say, to add the block number here?

Reply:
Hmm, not really I suppose, as it is going to be the identical too for the duration of one block. If the inherents are available, we could add the timestamp, though.

Reply:
> as it is going to be the identical too for the duration of one block

True, but being able to make the same sudo calls once per block is better than not being able to do them at all :thinking:

Reply:
Hmm, doesn't the current implementation already allow one unique sudo call? Realistically, the tx-pool will always be empty, as only sudo can send extrinsics anyhow with this signed extension.

So in practice, I think it is: allowing one unique call == allowing one unique call per block.

Reply:
> but for chains that also have the CheckNonce

So I assume that chains w/o `CheckNonce` are rather for development purpose. And typically runtime should provide other mean to distinguish transaction sent from the same account, not necessarily responsibility of the `sudo` itself.

We could use hash of call to avoid sending entire call across a runtime boundary, also txpool will have smaller amount of data to track.

Reply:
Someone would need to send the *exact* same *call* multiple times. Just sending multiple `transfer` calls is not the same. Each call has the arguments directly "baked in".

Reply:
Yes, I meant exactly the same call - for transfer that would be same amount, same address.

Reply:
I see this happening quite unlikely.